### ISSUE TITLE
Simplify test in this project

### ISSUE THREAD

**Issue by 304NotModified** [nunit/nunit.analyzers#70]
Hi,

I was trying to add a new fix + test, but IMO adding a testcase in this project is pretty complicated

In my opinion:

1. The test classes names are really too long
![image](https://user-images.githubusercontent.com/5808377/44313662-a8999380-a40c-11e8-8b10-8650e24aa1f4.png)

1. The tests in `TestCaseUsageAnalyzerTests` are using to much `nameof` 
1. The `BasePath` in `TestCaseUsageAnalyzerTests` is a bit of magic
1. The testcases are duplicates and hard to read, even in the simple case:
![image](https://user-images.githubusercontent.com/5808377/44313675-f6ae9700-a40c-11e8-8cc8-0ad675ba313d.png)

1. The `RunAnalysisAsync` should not accept an `Action<ImmutableArray<Diagnostic>>`, but a expectedDiagnostics (of type `IEnumerable<Diagnostic>`)

**Comment by mikkelbu** [comment]
Hi @304NotModified 
I wholeheartedly agree that the tests are too difficult to read, both due to the technical details, but also since the data and the assertions are separated into separate files.

I don't know the original reason for this, but I much prefer the solutions chosen in other roslyn-projects  where the data is just strings inside the test. Of cause one loses the intellisense, but this is still better than the current solution. I don't think there (currently) are any best practises for writing tests of analyzers/code fixes, but I know that @JohanLarsson has Gu.Roslyn.Asserts, and MS have several different variants (see e.g. https://github.com/dotnet/roslyn-analyzers/issues/1574#issuecomment-392066890).

I had planned to look into this after the first milestone, but perhaps we should take he battle now (so far I've lived with the pain in the few contributions I've made).

**Comment by 304NotModified** [comment]
Hi @mikkelbu 

Thanks for your reaction.

I do agree that strings are better, but I think there could be some improvement with the current approach (splitted testcases from tests). 

If you like, I will try to refactor a bit so it's more readable.

**Comment by 304NotModified** [comment]
made #73, please give your opinion :)

**Comment by JohanLarsson** [comment]
@304NotModified using Gu.Roslyn.Asserts tests look like [this](https://github.com/GuOrg/Gu.Analyzers/blob/master/Gu.Analyzers.Test/GU0082IdenticalTestCaseTests/Diagnostics.cs#L13-L31):

```cs
        [Test]
        public void TestCaseAttributeAndParameter()
        {
            var testCode = @"
namespace RoslynSandbox
{
    using NUnit.Framework;
    public class FooTests
    {
        [↓TestCase(1, 2)]
        [↓TestCase(1, 2)]
        public void Test(int i, int j)
        {
        }
    }
}";
            AnalyzerAssert.Diagnostics(Analyzer, ExpectedDiagnostic, testCode);
        }
```

**Comment by 304NotModified** [comment]
I think this also neat, and it's easier to work with compared to strings:

in #73

```
        [Test]
        public async Task CanAssignToWhenArgumentIsImplicitlyTypedArrayAndNotAssignable()
        {
            // option 2
            var (syntax, symbol, model) = await GetAttributeSyntaxAsync<CanAssignToWhenArgumentIsImplicitlyTypedArrayAndNotAssignable>();
            Assert.That(syntax.CanAssignTo(symbol, model), Is.False);
        }
```

**Comment by mikkelbu** [comment]
Hi both
I've been quite busy the last week, hence the silence. But I hope that I'll have time to look into this issue during the weekend or in the beginning of next week.
